lib/reports/resolvers.ts

```diff
@@ -244,3 +246,80 @@ export const generateApplicationsReport: Resolver<
     ok: true,
   };
 };
+
+/**
+ * Generates csv with accountants' info, given a start date and end date
+ * @returns Whether a csv could be generated (ok), and in the future an AWS S3 file link
+ */
+export const generateAccountantReport: Resolver<
+  QueryGenerateAccountantReportArgs,
+  GenerateAccountantReportResult
+> = async (_, args, { prisma }) => {
+  const {
+    input: { startDate, endDate },
+  } = args;
+
+  const paymentMethodGroups = await prisma.application.groupBy({
+    by: ['paymentMethod'],
+    where: {
+      createdAt: {
+        gte: startDate,
+        lte: endDate,
+      },
+    },
+    _sum: {
+      processingFee: true,
+      donationAmount: true,
+    },
+    _count: {
+      paymentMethod: true,
+    },
+  });
+
+  const totalAggregate = await prisma.application.aggregate({
+    _sum: {
+      processingFee: true,
+      donationAmount: true,
+    },
+    _count: {
+      paymentMethod: true,
+    },
+  });
+
+  const csvAccountantReportRows = [];
+  for (const key in paymentMethodGroups) {
+    const paymentMethodGroup = paymentMethodGroups[key];
+    csvAccountantReportRows.push({
+      rowName: formatPaymentType(paymentMethodGroup.paymentMethod),
+      countIssued: paymentMethodGroup._count.paymentMethod,
+      processingFee: paymentMethodGroup._sum.processingFee,
+      donationAmount: paymentMethodGroup._sum.donationAmount,
+      totalAmount: paymentMethodGroup._sum.donationAmount?.add(
+        paymentMethodGroup._sum.processingFee || 0
+      ),
```

Comment:
```suggestion
      totalAmount: Prisma.Decimal.add(
        paymentMethodGroup._sum.donationAmount || 0, 
        paymentMethodGroup._sum.processingFee || 0,
      )
```
Make sure to add `import { Prisma } from '@prisma/client'` at the top of the file to have access to `Prisma.Decimal`. We are using `Prisma.Decimal` for better precision when adding monetary values (eg. with normal computer arithmetic we get 0.1 + 0.2 != 0.3). Since either donation amount or processing fee could be `null`, we should use the `Prisma.Decimal` method `add` in the way I suggested above.

In the current code, if `donationAmount` is `null`, `totalAmount` will automatically be `null` since the `.add` won't get evaluated.

See docs: https://mikemcl.github.io/decimal.js/#Dadd

Reply:
This might not seem like an issue now, but if somehow Prisma breaks or we receive no donations in some timespan, we don't want to have the report show that processing fee was also 0